pymc3/distributions/continuous.py

```diff
@@ -1389,3 +1390,71 @@ def random(self, point=None, size=None, repeat=None):
     def logp(self, value):
         scaled = (value - self.mu) / self.beta
         return bound(-scaled - tt.exp(-scaled) - tt.log(self.beta), self.beta > 0)
+
+class Interpolated(Continuous):
+    R"""
+    Probability distribution defined as a linear interpolation of
+    of a set of points and values of probability density function
+    evaluated on them.
+
+    The points are not variables, but plain array-like objects, so
+    they are constant and cannot be sampled.
+
+    ========  =========================================
+    Support   :math:`x \in [x_points[0], x_points[-1]]`
+    ========  =========================================
+
+    Parameters
+    ----------
+    x_points : array-like
+        A monotonically growing list of values
+    pdf_points : array-like
+        Probability density function evaluated at points from `x`
+    """
+
+    def __init__(self, x_points, pdf_points, transform='interval',
+                 *args, **kwargs):
+        if transform == 'interval':
+            transform = transforms.interval(x_points[0], x_points[-1])
+        super(Interpolated, self).__init__(transform=transform,
+                                           *args, **kwargs)
+
+        interp = InterpolatedUnivariateSpline(x_points, pdf_points, k=1, ext='zeros')
```

Comment:
There are two major problems with non-linear interpolations:

1. PDF can become negative for higher-order spline interpolation at some points, it is nonsensical and can 
2. `random()` method can be implemented efficiently only for linear interpolation because inverse CDF can be expressed in closed form only for piecewise-quadratic CDF (well, it is possible to _try_ to do the same for piecewise-cubic CDF using Cardano formula, but I'm not subscribing to it :) For higher-order polynomial interpolations it would be necessary to find inverses numerically, using an iterative process like Newton method.